File: Classwork/dataStructures/src/timeTravel/TimeTravel.java
package timeTravel;

import static java.time.ZonedDateTime.now;
import static java.util.Arrays.stream;
import static java.util.TimeZone.getTimeZone;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class TimeTravel {
	public static void main(String[] args) {
		try (var fileReader = new Scanner(new File("Classwork/dataStructures/src/timeTravel/TimeTravel.txt"))) {
			var timeZoneID = getTimeZone("America/New_York").toZoneId();
			for (var i = 1; fileReader.hasNextLine(); i++) {
				var data = stream(fileReader.nextLine().split(" ")).mapToInt(Integer::parseInt).toArray();
				var currentTime = now(timeZoneID);
				var futureTime = currentTime.plusDays(data[0]).plusHours(data[1]).plusMinutes(data[2]);
				
			}
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		}
	}
}

File: Classwork/dataStructures/src/timeTravel/TimeTravel.kt
package timeTravel

import java.io.File
import java.io.FileNotFoundException
import java.time.ZonedDateTime.now
import java.time.format.DateTimeFormatter.RFC_1123_DATE_TIME
import java.util.*
import java.util.Arrays.stream
import java.util.TimeZone.getTimeZone

fun main() = try {
	Scanner(File("Classwork/dataStructures/src/timeTravel/TimeTravel.txt")).use {
		val timeZoneID = getTimeZone("America/New_York").toZoneId()
		var i = 1
		while (it.hasNextLine()) {
			val dataSet = stream(it.nextLine().split(" ").toTypedArray()).mapToInt(Integer::parseInt).toArray()
			val currentTime = now(timeZoneID)
			val futureTime = currentTime.plusDays(dataSet[0].toLong()).plusHours(dataSet[1].toLong())
				.plusMinutes(dataSet[2].toLong())
			println(
				"Trip $i:\n\tDeparture Date and Time: ${currentTime.format(RFC_1123_DATE_TIME)}\n\tArrival Date and Time: ${
					futureTime.format(
						RFC_1123_DATE_TIME
					)
				}"
			)
			i++
		}
	}
} catch (e: FileNotFoundException) {
	e.printStackTrace()
}
